src/views/forms/AccountManagementForms/WithdrawForm.tsx

```diff
@@ -101,50 +111,71 @@ export const WithdrawForm = () => {
             value: debouncedAmount,
             field: TransferInputField.usdcSize,
           });
-          setError(null);
+          setError(undefined);
         }
       } catch (error) {
-        setError(error);
+        setError(error.message);
       } finally {
         setIsLoading(false);
       }
     };
 
     setTransferValue();
-  }, [debouncedAmountBN.toNumber()]);
+  }, [debouncedAmountBN]);
+
+  const { screenAddresses } = useDydxClient();
+  const { dydxAddress } = useAccounts();
 
   const onSubmit = useCallback(
     async (e: FormEvent) => {
       try {
         e.preventDefault();
 
-        if (!requestPayload?.data || !debouncedAmountBN.toNumber()) {
+        if (!requestPayload?.data || !debouncedAmountBN.toNumber() || !toAddress || !dydxAddress) {
           throw new Error('Invalid request payload');
         }
 
         setIsLoading(true);
-        const txHash = await sendSquidWithdraw(debouncedAmountBN.toNumber(), requestPayload?.data);
-
-        if (txHash?.hash) {
-          const hash = `0x${Buffer.from(txHash.hash).toString('hex')}`;
-
-          addTransferNotification({
-            txHash: hash,
-            fromChainId: TESTNET_CHAIN_ID,
-            toChainId: chainIdStr || undefined,
-            toAmount: debouncedAmountBN.toNumber(),
-            triggeredAt: Date.now(),
-          });
-          abacusStateManager.clearTransferInputValues();
-          setWithdrawAmount('');
+        setError(undefined);
+
+        const screenResults = await screenAddresses({
+          addresses: [toAddress, dydxAddress],
+        });
+
+        if (screenResults?.[toAddress]) {
+          setError(
+            stringGetter({
+              key: STRING_KEYS.WALLET_RESTRICTED_WITHDRAWAL_TRANSFER_DESTINATION_ERROR_MESSAGE,
+            })
+          );
+        } else if (screenResults?.[dydxAddress]) {
+          setError(
+            stringGetter({
+              key: STRING_KEYS.WALLET_RESTRICTED_WITHDRAWAL_TRANSFER_ORIGINATION_ERROR_MESSAGE,
+            })
+          );
```

Comment:
is this the right priority? showing ur dydx address is restricted before or after the destination address?